ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/json/JsonExtractString.java

```diff
@@ -48,10 +46,10 @@ public String extract(
       return null;
     }
 
-    if (tokens == null) {
-      final JsonPathTokenizer tokenizer = new JsonPathTokenizer(path);
-      tokens = ImmutableList.copyOf(tokenizer);
-    }
+
+    final JsonPathTokenizer tokenizer = new JsonPathTokenizer(path);
+    final List<String> tokens = ImmutableList.copyOf(tokenizer);
```

Comment:
It should be good to keep a the `tokens` initialized once in case the `path` is equals to the latest path. What about adding a new class variable to keep the latest path seen, and check if the new path is equals to the latest path, if true, then re-use the current `tokens` class variable, if not, then initialize it with a new JsonPathTokenizer and List objects?

Reply:
ok, thanks for the feedback. I will cover your points.

Reply:
Done